### ISSUE TITLE
Lab3 peer review by Enrico Magliano

### ISSUE THREAD

**Issue by EnricoMagliano** [Diegomangasco/Computational_Intelligence#5]
<p>
In general the code is well organized and  the readme file contains usefull explainations, while the commets in the code  are poorly.
</p>
<p>
The evaluation is made using only one match, this is affect by randomness that can result in misleading results. Is better use more match and the win/number_of_match ratio (switching the starter).
</p>
<p>
All the solutions are made without considering the parameter k.
</p>
<h3>Task 1</h3>
<p>
Nothing to say, good implementations of the optimal strategy.
</p>
<h3>Task 2</h3>
<p>
Good hard coded strategy evolved by GA, the fitness is correct, in general all the structure of the evolution is correct except for the mutation that change the entire genome, while it should change only a part.
</p>
<h3>Task 3</h3>
<p>
Also here nothing to say, I really like your recursive function to explore the tree without directly create it.
</p>
<h3>Task 4</h3>
<p>
Good code structure of the evolved strategy, but I didn't understand how the reward are updates and why randomly initialize the rewards? 
</p>

<p>
In conclusion I can say that this work is reasoned and well implemented.

Good job.
</p>
